### PR TITLE
Feature/ar 156 update data its part1

### PR DESCRIPTION
**Pull request by rbradley813**

### CODE REVIEW

File: integration/src/main/java/dev/ikm/tinkar/integration/DataIntegrity.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static void startup(File datastoreRootLocation){
         LOG.info("Datastore location: " + datastoreRootLocation);
         CachingService.clearAll();
         ServiceProperties.set(ServiceKeys.DATA_STORE_ROOT, datastoreRootLocation);
-        PrimitiveData.selectControllerByName(TestConstants.SA_STORE_OPEN_NAME);
+        PrimitiveData.selectControllerByName("Open SpinedArrayStore");
```
Comment (dettmoney) [inline integration/src/main/java/dev/ikm/tinkar/integration/DataIntegrity.java:78]: Why are we going away from using a constant here?

File: integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java [hunk 2]
```diff
@@ -16,30 +16,17 @@
 package dev.ikm.tinkar.integration.langext.fhir;
 
 import dev.ikm.tinkar.common.id.IntIds;
-import dev.ikm.tinkar.coordinate.stamp.StampCoordinateRecord;
-import dev.ikm.tinkar.coordinate.stamp.StampCoordinateRecordBuilder;
-import dev.ikm.tinkar.coordinate.stamp.StampPositionRecord;
-import dev.ikm.tinkar.coordinate.stamp.StampPositionRecordBuilder;
-import dev.ikm.tinkar.coordinate.stamp.StateSet;
+import dev.ikm.tinkar.coordinate.stamp.*;
```
Comment (dettmoney) [inline integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java:19]: We should avoid using * in import statements

File: integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java [hunk 3]
```diff
@@ -16,30 +16,17 @@
...
 import dev.ikm.tinkar.coordinate.stamp.calculator.StampCalculator;
 import dev.ikm.tinkar.coordinate.stamp.calculator.StampCalculatorWithCache;
-import dev.ikm.tinkar.entity.ConceptEntity;
-import dev.ikm.tinkar.entity.ConceptEntityVersion;
-import dev.ikm.tinkar.entity.Entity;
-import dev.ikm.tinkar.entity.EntityService;
-import dev.ikm.tinkar.entity.EntityVersion;
-import dev.ikm.tinkar.entity.SemanticEntity;
+import dev.ikm.tinkar.entity.*;
```
Comment (dettmoney) [inline integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java:22]: We should avoid * in import statements

File: integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java [hunk 4]
```diff
@@ -16,30 +16,17 @@
...
+import dev.ikm.tinkar.integration.helper.DataStore;
 import dev.ikm.tinkar.integration.helper.TestHelper;
 import dev.ikm.tinkar.terms.TinkarTerm;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Disabled;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.*;
```
Comment (dettmoney) [inline integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java:29]: Avoid * in imports

File: integration/src/test/java/dev/ikm/tinkar/integration/snomed/core/QueryByMembershipIT.java [hunk 5]
```diff
@@ -15,27 +15,14 @@
...
-import dev.ikm.tinkar.common.service.ServiceProperties;
-import dev.ikm.tinkar.entity.ConceptEntity;
-import dev.ikm.tinkar.entity.Entity;
-import dev.ikm.tinkar.entity.EntityService;
-import dev.ikm.tinkar.entity.EntityVersion;
-import dev.ikm.tinkar.entity.PatternEntity;
-import dev.ikm.tinkar.entity.SemanticEntity;
-import dev.ikm.tinkar.entity.StampEntity;
-import dev.ikm.tinkar.entity.StampEntityVersion;
+import dev.ikm.tinkar.entity.*;
```
Comment (dettmoney) [inline integration/src/test/java/dev/ikm/tinkar/integration/snomed/core/QueryByMembershipIT.java:19]: import * issue

File: integration/src/test/java/dev/ikm/tinkar/integration/snomed/core/QueryByMembershipIT.java [hunk 6]
```diff
@@ -15,27 +15,14 @@
...
+import dev.ikm.tinkar.integration.helper.DataStore;
+import dev.ikm.tinkar.integration.helper.TestHelper;
 import dev.ikm.tinkar.schema.TinkarMsg;
 import dev.ikm.tinkar.terms.TinkarTerm;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Disabled;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Order;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (dettmoney) [inline integration/src/test/java/dev/ikm/tinkar/integration/snomed/core/QueryByMembershipIT.java:25]: import * issue

File: integration/src/main/java/dev/ikm/tinkar/integration/DataIntegrity.java (hunk 1 above)
Comment (rbradley813) [reply integration/src/main/java/dev/ikm/tinkar/integration/DataIntegrity.java:78]: @dettmoney, the constant was removed in preference of an enum within the test structure. This DataIntegrity class should not live here and will be moved elsewhere in another 'part' of this update to the data integration tests.

File: integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java (hunk 2 above)
Comment (rbradley813) [reply integration/src/test/java/dev/ikm/tinkar/integration/langext/fhir/FhirTransformAPIIT.java:19]: @dettmoney, thank you for the 'offline' suggestion about increasing the import * threshold for the IntelliJ Optimize Import settings.

File: integration/src/test/java/dev/ikm/tinkar/integration/integrity/DataIntegrityIT.java [hunk 7]
```diff
@@ -19,13 +19,16 @@
 import dev.ikm.tinkar.common.id.IntIdSet;
 import dev.ikm.tinkar.common.id.IntIds;
 import dev.ikm.tinkar.common.id.PublicId;
-import dev.ikm.tinkar.common.service.*;
+import dev.ikm.tinkar.common.service.PrimitiveData;
+import dev.ikm.tinkar.common.service.TinkExecutor;
+import dev.ikm.tinkar.common.util.io.FileUtil;
 import dev.ikm.tinkar.entity.*;
```
Comment (dettmoney) [inline integration/src/test/java/dev/ikm/tinkar/integration/integrity/DataIntegrityIT.java:25]: .* import

File: integration/src/test/java/dev/ikm/tinkar/integration/provider/spinedarray/OwlTransformationTestIT.java [hunk 8]
```diff
@@ -15,13 +15,13 @@
  */
 package dev.ikm.tinkar.integration.provider.spinedarray;
 
-import dev.ikm.tinkar.common.service.PrimitiveData;
 import dev.ikm.tinkar.entity.*;
```
Comment (dettmoney) [inline integration/src/test/java/dev/ikm/tinkar/integration/provider/spinedarray/OwlTransformationTestIT.java:18]: .* import

File: integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/EphemeralProtobufIT.java [hunk 9]
```diff
@@ -23,37 +23,42 @@
...
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
-public class EphemeralProtobufIT extends TestHelper {
+public class EphemeralProtobufIT {
 
     private static final Logger LOG = LoggerFactory.getLogger(EphemeralProtobufIT.class);
 
     @BeforeAll
-    public void setupSuite() {
-       startEphemeralDataBase();
+    public static void beforeAll() {
```
Comment (stenkarl) [inline integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/EphemeralProtobufIT.java:46]: I don't think this needs to be static.

File: integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/OwlToLogicalAxiomIT.java [hunk 10]
```diff
@@ -36,12 +46,18 @@
...
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
-public class OwlToLogicalAxiomIT extends TestHelper {
+public class OwlToLogicalAxiomIT {
     private static final Logger LOG = LoggerFactory.getLogger(OwlToLogicalAxiomIT.class);
 
     @BeforeAll
-    public static void setUp() {
-        loadEphemeralDataBase();
+    public static void beforeAll() {
```
Comment (stenkarl) [inline integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/OwlToLogicalAxiomIT.java:53]: Same here. Doesn't need to be static.

File: integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/OwlToLogicalAxiomIT.java [hunk 11]
```diff
@@ -36,12 +46,18 @@
...
+        TestHelper.startDataBase(DataStore.EPHEMERAL_STORE);
+        TestHelper.loadDataFile(TestConstants.PB_STARTER_DATA_REASONED);
+    }
+
+    @AfterAll
+    public static void afterAll() {
```
Comment (stenkarl) [inline integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/OwlToLogicalAxiomIT.java:59]: And here. Doesn't need to be static.

File: integration/src/test/java/dev/ikm/tinkar/integration/snomed/core/QueryByMembershipIT.java [hunk 12]
```diff
@@ -51,44 +38,30 @@
...
 
-    public void initialize(File datastore) throws IOException {
-        LOG.info("Starting database");
-        LOG.info("Loading data from " + DATA_STORE_FILE.getAbsolutePath());
-        CachingService.clearAll();
-        ServiceProperties.set(ServiceKeys.DATA_STORE_ROOT, datastore);
-        PrimitiveData.selectControllerByName(CONTROLLER_NAME);
-        PrimitiveData.start();
+    @BeforeAll
+    public static void beforeAll() {
```
Comment (stenkarl) [inline integration/src/test/java/dev/ikm/tinkar/integration/snomed/core/QueryByMembershipIT.java:57]: Another unnecessary static

File: integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/EphemeralProtobufIT.java (hunk 9 above)
Comment (rbradley813) [reply integration/src/test/java/dev/ikm/tinkar/integration/provider/ephemeral/EphemeralProtobufIT.java:46]: @stenkarl, as discussed JUnit enforces static for this typeof TestInstance.Lifecycle configuration. We expect to reconsider this as we mature our testing architecture, especially as we reorganize integration tests and implement test suites.

**Review by stenkarl — APPROVED** [review]

**Pull request merged**